spec/Vehicle/Connectivity.vspec

```diff
@@ -21,3 +21,130 @@ IsConnectivityAvailable:
   comment: This signal can be used by onboard vehicle services to decide what features that
            shall be offered to the driver, for example disable the 'check for update' button if vehicle does
            not have connectivity.
+           
+ConnectivityType:
+  datatype: string[]
+  type: attribute
+  allowed: [
+    'CELLULAR',
+    'WIFI'
+    ]
+  description: Type of connectivity currently available.
+  
+CurrentDownload:
+  datatype: float
+  type: sensor
+  unit: MiB/s
+  description: Download bandwidth currently in-use.
+
+CurrentUpload:
+  datatype: float
+  type: sensor
+  unit: MiB/s
+  description: Upload bandwidth currently in-use.
+
+Cellular:
+  type: branch
+  description: Properties related to cellular connectivity.
+  
+Cellular.SignalStrength:
+  type: sensor
+  datatype: int8
+  unit: dBm
+  description: Cellular connectivity signal strength.
```

Comment:
This is defined as int8 (-128 to 127) for both WiFi and cellular. Given that dBm is a logarithmic scale, I didn't see a scenario where the signal strength could stray outside of this range, but it is a bit tight. Alternatively, would this be better as a percentage (0-100%)?

Reply:
We also have float as an option. Historically in VSS when selecting datatype we typically consider expected range and if decimal notation is required. If we see it unlikely that someone needs decimals for the given unit we select an int-type that fits what we see as a reasonable range.  (What datatype that then is used in a real deployment could differ).

Concerning using percentage - In user interfaces you rarely see the absolute value but rather a relative value like number of "bars" where 4 bars may indicate excellent signal strength and 1 bar bad signal strength. But I do not know if there is any industry standard or convention converting between dBM and percentage/bars or if that is only in relationship to what the device expects. I.e. X dBM might mean 40% for one device but 20% for another device?

In the long run maybe we need both an absolute signal (dBM) and a relative signal (starts/bars/percent) as they might serve different purposes. Maybe the OEM want to collect absolute dBM values, but the HMI wants to show the relative value. If we start with absolute, if that fits your use case, we could always add relative later.

Reply:
I can see value to both absolute and relative, for sure. I think absolute could be useful for things like vehicle diagnostics and troubleshooting.

There isn't really a standard conversion between dBm and percentage, but I think most implementations are at least similar. Because actual connectivity drops off much faster at low connections, many companies actually choose to drop percentage faster as they get to the lower range, so it isn't always linear either. I think I'll revise this part to include both.